File: backendApp/src/main/java/com/cinemas/entities/Review.java
package com.cinemas.entities;

import com.cinemas.enums.ReviewType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
@Table(name = "review")
public class Review {
    @Id
    @GeneratedValue
    private Integer id;

    @Column
    private String name;

    @Column
    private ReviewType type;

    @Column
    private Integer view;

    @Column
    private String slug;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column
    private String thumbnail;

    @ManyToOne
    @JoinColumn(name = "movie_id")
    @JsonIgnore
    private Movie movie;
}

File: fllmbooking_app/lib/data/models/review.dart
class Review {
  int? id;
  String? name;
  String? slug;
  String? description;
  String? thumbnail;
  int? view;
  // ReviewType? type;

  Review({
    this.id,
    this.name,
    this.slug,
    this.description,
    this.thumbnail,
    this.view,
    // this.type,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      name: json['name'],
      slug: json['slug'],
      description: json['description'],
      thumbnail: json['thumbnail'],
      view: json['view'],
      // type: json['type'],
    );
  }
}
